pkg/capabilities/triggers/mercury_trigger.go

```diff
@@ -0,0 +1,172 @@
+package triggers
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+// TODO: Register the capiability in the NewDelegateFunction in chalink/core/services/workflows.delegate.go
+
+var mercuryInfo = capabilities.MustNewCapabilityInfo(
+	"mercury-trigger",
+	capabilities.CapabilityTypeTrigger,
+	"An example mercury trigger.",
+	"v1.0.0",
+)
+
+// This Trigger Service allows for the registration and deregistration of triggers. You can also send reports to the service.
+type MercuryTriggerService struct {
+	capabilities.CapabilityInfo
+	chans               map[workflowID]chan<- capabilities.CapabilityResponse
+	feedIdsForTriggerId map[string][]uint64
+	mu                  sync.Mutex
+	feedIdToWorkflowId  map[uint64]string
+}
+
+type MercuryReport struct {
+	feedId               uint64
+	fullreport           []byte
+	benchmarkPrice       int64
+	observationTimestamp uint32
+}
+
+type MercuryTriggerEvent struct {
+	triggerType string // "mercury"
+	id          string // "sha256 of payload.feedId + payload.timestamp"
```

Comment:
I left this comment on the design doc too - I believe that a trigger event needs to contain multiple reports. Otherwise we won't be able to batch feed updates in onchain transactions - workflow executions are independent.

In consequence, I don't think that trigger ID can be based on a single feedID.